addons/point_of_sale/models/pos_session.py

```diff
@@ -581,12 +581,17 @@ def _create_invoice_receivable_lines(self, data):
 
         invoice_receivable_vals = defaultdict(list)
         invoice_receivable_lines = {}
-        for receivable_account_id, amounts in invoice_receivables.items():
-            invoice_receivable_vals[receivable_account_id].append(self._get_invoice_receivable_vals(receivable_account_id, amounts['amount'], amounts['amount_converted']))
-        for receivable_account_id, vals in invoice_receivable_vals.items():
+        for partner, amounts in invoice_receivables.items():
+            account_id = partner.property_account_receivable_id.id
+            invoice_receivable_vals[partner].append(self._get_invoice_receivable_vals(account_id, amounts['amount'], amounts['amount_converted'], partner=partner))
```

Comment:
Not sure, but perhaps here we should use the `partner.commercial_partner_id`, otherwise when invoicing from POS, we may have the INV entry assigned to the parent company and the POS entry to the child, unbalancing both partner ledgers

Reply:
I'm also thinking the same, but when calling `_prepare_invoice_vals` for the creation of invoice, we are using only the [partner_id](https://github.com/odoo/odoo/blob/8da349e785b5b461ea19874df9686f8cd26af8ad/addons/point_of_sale/models/pos_order.py#L473).

But if the account_move creation is finding the commercial partner, then we should use the commercial partner as well for the creation of invoice lines.

Reply:
Looks like the commercial partner is automatically set as partner_id of the created account move: https://github.com/odoo/odoo/blob/3b199007bd5a472d57f823a63b2b77841b14c7fc/addons/account/models/account_move.py#L1627